src/v/cluster/topics_frontend.h

```diff
@@ -180,6 +183,20 @@ class topics_frontend {
       do_cancel_moving_partition_replicas(
         std::vector<model::ntp>, model::timeout_clock::time_point);
 
+    struct health_monitor_info {
+        absl::flat_hash_map<model::node_id, node_disk_space> node_disk_reports;
+        absl::flat_hash_map<model::partition_id, int64_t> ntp_sizes;
```

Comment:
`ntp_sizes` will be huge on topics with tens of thousands of partitions, does this really need to individually record each partition's?

Reply:
Yes, We need partition info to pass it into constrains inside `partition_allocator`